notification-bot/src/index.ts

```diff
@@ -0,0 +1,215 @@
+
+    const provider = new RpcProvider({
+      nodeUrl: constants.NetworkName.SN_SEPOLIA,
+      chainId: constants.StarknetChainId.SN_SEPOLIA,
+    });
+
+    const hashAndBlockNumber = await provider.getBlockLatestAccepted();
+    const block_number = hashAndBlockNumber.block_number;
+
+    //Initialize the filter
+    const filter_test = initializeFilter();
+
+    // Configure the apibara client
+    configureClient(client, filter_test, block_number);
+    await listenToMessages(client);
+
+  } catch (error) {
+    console.error("Initialization failed", error);
+    process.exit(1);
+  }
+}
+
+/**
+ * Listens to blockchain events and handles each submit proposel event as they are received.
+ * @param {StreamClient} client The Apibara client instance from which to listen for messages.
+ */
+async function listenToMessages(client: StreamClient) {
+  for await (const message of client) {
+    if (message.message === "data" && message.data?.data) {
+      for (const data of message.data.data) {
+        const block = v1alpha2.Block.decode(data);
+
+        const { header, events, transactions } = block;
+
+        if (!header || !transactions) {
+          continue;
+        }
+
+        for (const event of events) {
+          console.log(event.event)
+          if (event.event && event.event.keys && event.event.data) {
+            for (let evtkey of event.event!.keys) {
+              let evtkey_hex = FieldElement.toHex(evtkey);
+
+              let submit_proposal_hash = PROPOSAL_HASH;
+              let isSame = evtkey_hex === submit_proposal_hash;
+
+              if (isSame) {
+                handleEventSubmitProposal(header, event.event);
+              }
+            }
+          }
+          break;
+        }
+      }
+    }
+  }
+}
+
+
+main()
+  .then(() => process.exit(0))
+  .catch((error) => {
+    console.error(error);
+    process.exit(1);
+  });
+
+/**
+ * Converts a hexadecimal string to a decimal string.
+ * @param {string | number} hex The hexadecimal string or number to convert.
+ * @returns {string} The decimal string representation of the hex input.
+ */
+function hexToDecimalString(hex: string | number): string {
+  const hexString = hex.toString().startsWith('0x') ? hex.toString().substring(2) : hex.toString();
+
+  const decimalNumber = parseInt(hexString, 16);
+
+  return decimalNumber.toString();
+}
+
+/**
+ * Handles individual blockchain events, and notifies user for proposal submission events.
+ * @param {v1alpha2.IBlockHeader} header The header of the blockchain block containing the event.
+ * @param {v1alpha2.IEvent} event The event to process.
+ */
+async function handleEventSubmitProposal(
+  header: v1alpha2.IBlockHeader,
+  event: v1alpha2.IEvent,
+) {
+  console.log("STARTING TO HANDLE PROPOSAL");
+
+  console.log(event);
+
+  const sender = event.fromAddress ? FieldElement.toHex(event.fromAddress) : null;
+  const payload = event.data ? FieldElement.toHex(event.data[1]) : null;
```

Comment:
If there are no data in the event, it's probably an error, it should output an error message (or send it to Telegram).

Reply:
done